Source/WebCore/rendering/RenderObject.cpp

```diff
@@ -2282,6 +2282,13 @@ static Vector<FloatRect> absoluteRectsForRangeInText(const SimpleRange& range, T
         return { };
 
     auto offsetRange = characterDataOffsetRange(range, node);
+    // Move to surrogate pair start for Range start and past surrogate pair end for Range end in case the trailing surrogate is indexed.
+    ASSERT(offsetRange.start < node.data().length());
+    if (U16_IS_TRAIL(node.data()[offsetRange.start]) && offsetRange.start && U16_IS_LEAD(node.data()[offsetRange.start - 1]))
+        offsetRange.start--;
+    ASSERT(offsetRange.end < node.data().length());
```

Comment:
Ditto. Also, we need to clamp the end at the actual end of the current node since characterDataOffsetRange will return `std::numeric_limits<unsigned>::max()` when the node isn't the end container.

Reply:
I think we can avoid doing anything at the very start and end of ranges as we are testing for the case where we are indexing into the surrogate pair, so also no need to clamp.
I added a subtest where the range consists of a different start and end text container to test the case where std::numeric_limits<unsigned>::max() is returned.